### DISCUSSION TITLE
Clearing date field result in error SQL update error

### DISCUSSION THREAD

**Discussion by mmerlijn** [livewire/livewire#4220] [1. Help] [answered]
Problem: Setting or updating a date property works fine. But after clearing the date field in the browser. An SQL insert error occured (empty string in date field).
In the payload the date field will be set to an empty string instead of null. Is this a bug or do I miss something?
Tested with Chrome and Firefox

I tried to check if the ```$this->client->dob``` is an empty string, but this becomes an Carbon object. 

Is there a way to cast the empty string to a null value? Or an other work around?

Simplification of the used code:

Model Client

```php
Class Client extends Model{
    protected $casts = ['dob' => 'date:Y-m-d'];
...
```

Livewire component

```php
class Client extends Component
{
    public Client $client;

    protected $rules = ['client.dob' => 'date|nullable'];

    public function updateClient()
    {
        $this->validate();
        $this->client->save();
    }
    ...
```

Livewire blade file

```html
<div>
    <input type="date" wire:model="client.dob">
</div>
```

**Comment by jonneroelofs** [comment, marked as answer]
I have used a mutator for this in the past. So on the Client model you could add something like:

```php
public function setDobAttribute($value){
        if(empty($value)){
            $this->attributes['dob'] = null;
        } else {
            $this->attributes['dob'] = $value;
        }
}
```

**Reply by mmerlijn** [reply]
Thnx for the quick response. This works fine
